### PR TITLE
Added support for running with different builder

### PR DESCRIPTION
**Pull request by ty2**
Introduced builder arguments to specify custom builder.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -135,6 +135,7 @@ Options:
   --cache-map    The map of actions source to container destination paths for the cache paths
   --scratch-dir  Where the action is stores some temporary files for its processing. Default: 'scratch'
   --skip-extraction  Skip the extraction of the cache from the docker container
+  --builder     The name of the builder. Default: 'default'
```
Comment (AkihiroSuda) [inline README.md:138]:
```suggestion
  --builder     The name of the buildx builder. Default: 'default'
```

Comment (AkihiroSuda) [reply README.md:138]: Also, please squash the commits

Comment (ty2) [reply README.md]: Done.

**Comment by AkihiroSuda** [comment]
Please rebase and rebuild the `dist`:
https://github.com/reproducible-containers/buildkit-cache-dance/blob/7c892679bab8ff382a8c88ab7f973d5e30a8f239/.github/workflows/unit-tests.yml#L26

**Comment by ty2** [comment]
> Please rebase and rebuild the `dist`:
> 
> https://github.com/reproducible-containers/buildkit-cache-dance/blob/7c892679bab8ff382a8c88ab7f973d5e30a8f239/.github/workflows/unit-tests.yml#L26
> 
> Also please squash the commits

Done.

**Comment by AkihiroSuda** [comment]
Commits do not seem squashed

https://www.freecodecamp.org/news/git-squash-commits/

File: pnpm-lock.yaml [hunk 2]
```diff
@@ -2872,4 +2872,4 @@ packages:
   /yocto-queue@1.0.0:
     resolution: {integrity: sha512-9bnSc/HEW2uRy67wc+T8UwauLuPJVn28jb+GtJY16iiKWyvmYJRXVT4UamsAEGQfPohgr2q4Tq0sQbQlxTfi1g==}
     engines: {node: '>=12.20'}
-    dev: true
+    dev: true
```
Comment (AkihiroSuda) [inline pnpm-lock.yaml:2875]: Unnecessary change

**Comment by ty2** [comment]
Sorry, I'm not familiar with Git. It looks like the commits have been squashed. Could you check if there are any issues?

**Review by AkihiroSuda — COMMENTED** [review]
Thanks

**Comment by kkopachev** [comment]
Hardcoding `default` builder name would break github actions setup with `setup-buildx-action` as it creates new builder with unique name and sets it as default, while the builder named `default` still exists, it is not selected for builds. When this action specifies `--builder default` it will use another builder, not the one for actual payload builds

**Comment by kkopachev** [comment]
I'd suggest using `BUILDX_BUILDER` environment variable instead of introducing new argument

**Comment by slinkydeveloper** [comment]
Stumbled on this in CI, used in combination with setup-buildx-action:

```
Run reproducible-containers/buildkit-cache-dance@v3

FROM ghcr.io/containerd/busybox:latest
COPY buildstamp buildstamp
RUN --mount=type=cache,target=/var/cache/sccache     --mount=type=bind,source=.,target=/var/dance-cache     cp -p -R /var/dance-cache/. /var/cache/sccache  || true

Error running command: docker buildx build --builder default -f scratch/Dancefile.inject --tag dance:inject sccache-cache
ERROR: use `docker --context=default buildx` to switch to context "default"
```

**Comment by slinkydeveloper** [comment]
Pinning the action version to 3.1.2 fixes it for me.
